books_manager.rb

```diff
@@ -0,0 +1,22 @@
+class BooksManager
+  attr_reader :books
+
+  def initialize(books)
+    @books = books
+  end
+
+  def list_books
+    @books.map { |book| puts "Title: '#{book.title}', Author: '#{book.author}'" }
+  end
+
+  def create_book
+    puts "Please enter the book's"
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    new_book = Book.new(title, author)
+    @books.push(new_book)
+    print "The book '#{new_book.title}' by '#{new_book.author}' was added successfully"
```

Comment:
Nice job so far😉. I noticed that both the `main` class,  `app` class 
<img width="323" alt="image" src="https://github.com/jlvFlores/OOP-school-library/assets/64643995/78430f1e-0112-474b-ba14-49f759bd83d2">
, `book_manager`, and `rental_manager` class share the responsibility of requesting and validating user inputs. To improve the code's maintainability and adhere to the Single Responsibility Principle (SRP) or SOLID principles, I would suggest you assign the responsibility of validating and requesting inputs to the `main` class😉. In implementing this, kindly take all prompts or commands to request and validate inputs to the `main` class and you can call them here where necessary😉.
By doing this, you'll achieve a clearer separation of concerns and improve the codebase's overall structure. Each class will have a single responsibility, making the code easier to understand, modify, and test.

you can take a look at [this lesson](https://www.digitalocean.com/community/conceptual-articles/s-o-l-i-d-the-first-five-principles-of-object-oriented-design) for a better understanding on the SOLID principle😉.

kindly note that the image above is from your `app` class😉.

Reply:
Hello @Ibizugbe thanks for the recommendation, It's helped me come to a better solution. I've decided to make the `app` class responsible for validating and requesting input, since having the `main` class be responsible for it would leave `app` without a purpose as `app` would only call external methods with the same name making the code redundant. At the same time I chose to give `main` the responsibility of deciding what method to execute. After doing this I finally had the `books, people and rentals manager` classes only storing their respective array and having unique clean and easy to read methods. 👍

Reply:
Also, you have some other people's names in your main comment, lol 😅

Reply:
nice job @jlvFlores  😉😉😉 , I am happy that was useful to you . and also, Thanks for pointing out that error😅 , I will correct it immediately😅😅😉😉